### PR TITLE
DO NOT MERGE: Hardware limitation testing

### PR DESCRIPTION
**Pull request by Sybrand**
TODO: Incorporate changes introduced in b8c274e359b5bd05fac7f2db36ca1a43e666bf62

# Goal
Leverage existing compose files to quickly spin up entire system in AWS and test hardware limitations for large datasets.

# Additions of note
- inclusion of minio
- inclusion of postgres
- inclusion of "utilty" container
- inclusion of pipeline container

# Changes of note
- web-renderer port changed due to conflict with minio

# Testing
The secret sauce to get this to run is the correct environment variables. Ping me and I'll send you my `.env` - otherwise you can work from the example env files.

Follow the instruction in the demonstration/README.md

# Future PR's
- Modifications to druid to ensure limiting memory usage (will be product of testing in AWS)
- Test/run in AWS
- Move code to BR fork

### CODE REVIEW

**Comment by AbbyBeeler** [comment]
@Ideonvn I can review for Brazil specific content, but can you take reviewing from a technical and best practices standpoint? I'm not familiar enough with docker composes to be very helpful there

**Review by Ideonvn — COMMENTED** [review]
@Sybrand is the idea of this PR to eventually merge or just for testing purposes?
If we do plan to merge it in the future, lets create it's own folder.

The current docker-compose file is purely for web prod & staging, prefer to keep it as such.

**Comment by Sybrand** [comment]
> @Sybrand is the idea of this PR to eventually merge or just for testing purposes? If we do plan to merge it in the future, lets create it's own folder.
> 
> The current docker-compose file is purely for web prod & staging, prefer to keep it as such.

The intention is just for testing, but that doesn't change that it probably has to get merged to be practical.

I can pull it into it's own file+folder - any preferences? (it does suck having so many different but almost identical compose files!)

**Comment by Ideonvn** [comment]
> > @Sybrand is the idea of this PR to eventually merge or just for testing purposes? If we do plan to merge it in the future, lets create it's own folder.
> > The current docker-compose file is purely for web prod & staging, prefer to keep it as such.
> 
> The intention is just for testing, but that doesn't change that it probably has to get merged to be practical.
> 
> I can pull it into it's own file+folder - any preferences? (it does suck having so many different but almost identical compose files!)

@Sybrand do take a look at [docker-compose extend](https://docs.docker.com/compose/extends/), believe this should be sufficient.

File: demonstration/harmony/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+    restart: always
+    pull_policy: always
+    ports:
+      # minio uses 9000, so we bumped it to 9001
+      - 9001:9000
+    environment:
+      - ALLOW_HTTP=true
+      - PORT=9000
+    networks:
+      - harmony_network
```
Comment (Ideonvn) [inline demonstration/harmony/docker-compose.yml:232]: @Sybrand do we know for sure if this is being used at all anymore?

**Review by Ideonvn — COMMENTED** [review]
Overall this looks good, going to spin this up locally today.
What are the overall hardware requirements for this? Will my local machine survive?

File: demonstration/harmony/docker-compose.yml (hunk 1 above)
Comment (Sybrand) [reply demonstration/harmony/docker-compose.yml:232]: No Idea - but could argue it makes sense to keep in "demo" - as it illustrates an example of having a container with a worker running.

Comment (Ideonvn) [reply demonstration/harmony/docker-compose.yml:232]: Well, afaik this was replaced by Urlbox... hence asking.

Comment (Sybrand) [reply demonstration/harmony/docker-compose.yml:232]: No idea. @AbbyBeeler  / @cbear99  you have any idea?

Comment (AbbyBeeler) [reply demonstration/harmony/docker-compose.yml:232]: Yeah it's not used anymore and could be removed

File: demonstration/harmony/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+  redis:
+    image: redis:latest
+    restart: always
+    pull_policy: always
+    ports:
+      - 6379:6379
+    networks:
+      - harmony_network
+
+  hasura:
```
Comment (AbbyBeeler) [inline demonstration/harmony/docker-compose.yml:243]: prod hasura starts differently than local dev. is this working? 

once everything is running, you'd want to run `scripts/data_catalog/populate_query_models_from_config.py --sql_connection_string <>` against the web container. and then AQT should load.

**Comment by Ideonvn** [comment]
Don't know why, but the there is now an urge to merge

**Pull request closed without merging**